homeassistant/components/climate/spider.py

```diff
@@ -0,0 +1,143 @@
+from homeassistant.components.spider import DOMAIN
+from homeassistant.const import TEMP_CELSIUS
+
+DEPENDENCIES = ['spider']
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Spider thermostat."""
+    thermostats = hass.data[DOMAIN]['thermostats']
+    api = hass.data[DOMAIN]['controller']
+    for i, thermostat in enumerate(thermostats):
+        add_devices(
+            [SpiderThermostat(api, thermostat['id'], thermostat['name'])],
+            True
+        )
+
+
+class SpiderThermostat(ClimateDevice):
+    """Representation of a thermostat."""
+
+    def __init__(self, client, thermostat_id, name):
+        """Initialize the thermostat."""
+        self.client = client
+        self._id = thermostat_id
+        self._name = name
+        self._master = False
+        self._thermostat = None
+        self._current_temperature = None
+        self._target_temperature = None
+        self._min_temp = None
+        self._max_temp = None
+        self._operation = STATE_IDLE
+        self._operation_list = [
+            STATE_HEAT,
+            STATE_COOL,
+        ]
+
+    @property
+    def supported_features(self):
+        """Return the list of supported features."""
+        supports = SUPPORT_TARGET_TEMPERATURE
+
+        if self._master is True:
+            supports = supports | SUPPORT_OPERATION_MODE
+
+        return supports
+
+    @property
+    def name(self):
+        """Return the name of the thermostat, if any."""
+        return self._name
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement."""
+        return TEMP_CELSIUS
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        return self._current_temperature
+
+    @property
+    def target_temperature(self):
+        """Return the temperature we try to reach."""
+        return self._target_temperature
+
+    @property
+    def min_temp(self):
+        """Return the minimum temperature."""
+        return self._min_temp
+
+    @property
+    def max_temp(self):
+        """Return the maximum temperature."""
+        return self._max_temp
+
+    @property
+    def current_operation(self: ClimateDevice) -> str:
+        """Return current operation ie. heat, cool, idle."""
+        return self._operation
+
+    @property
+    def operation_list(self):
+        """Return the list of available operation modes."""
+        return self._operation_list
+
+    def set_temperature(self, **kwargs):
+        """Set new target temperature."""
+        temperature = kwargs.get(ATTR_TEMPERATURE)
+        if temperature is None:
+            return
+
+        self.client.set_temperature(self._thermostat, temperature)
+
+    def set_operation_mode(self, operation_mode):
+        """Set new target operation mode."""
+        self.client.set_operation_mode(self._thermostat, operation_mode)
```

Comment:
Do we need to translate between home assistant operation mode and device mode?

Reply:
heat = Heat
cool = Cool

The API doesn't allow everything lowercased, so I found it better to capitalize the first character within the API lib itself. So no translation needed here.